  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # effort Controllers (axles) ---------------------------------------
  front_left_effort_controller:
    type: effort_controllers/JointEffortController
    joint: fl_wheel_axle
  back_left_effort_controller:
    type: effort_controllers/JointEffortController
    joint: bl_wheel_axle
  front_right_effort_controller:
    type: effort_controllers/JointEffortController
    joint: fr_wheel_axle
  back_right_effort_controller:
    type: effort_controllers/JointEffortController
    joint: br_wheel_axle


  # effort Controllers (swivels) ---------------------------------------
  front_left_swivel_controller:
    type: effort_controllers/JointEffortController
    joint: fl_swivel_rev
  back_left_swivel_controller:
    type: effort_controllers/JointEffortController
    joint: bl_swivel_rev
  front_right_swivel_controller:
    type: effort_controllers/JointEffortController
    joint: fr_swivel_rev
  back_right_swivel_controller:
    type: effort_controllers/JointEffortController
    joint: br_swivel_rev

  # shock controllers ----------------------------------------
  front_left_shock_controller:
    joint: fl_drive_shock
    type: effort_controllers/JointPositionController
    pid: {p: 10000, i: 0.0, d: 20}
  back_left_shock_controller:
    joint: bl_drive_shock
    type: effort_controllers/JointPositionController
    pid: {p: 10000, i: 0.0, d: 20}
  front_right_shock_controller:
    joint: fr_drive_shock
    type: effort_controllers/JointPositionController
    pid: {p: 10000, i: 0.0, d: 20}
  back_right_shock_controller:
    joint: br_drive_shock
    type: effort_controllers/JointPositionController
    pid: {p: 10000, i: 0.0, d: 20}
